source/common/runtime/runtime_features.cc

```diff
@@ -61,6 +61,7 @@ constexpr const char* runtime_features[] = {
     "envoy.reloadable_features.correct_scheme_and_xfp",
     "envoy.reloadable_features.disable_tls_inspector_injection",
     "envoy.reloadable_features.fix_added_trailers",
+    "envoy.reloadable_features.fix_locality_update_for_eds_cluster_endpoints",
```

Comment:
I know this follows the above `fix_added_trailers` (also naming is hard!), but can we have a better name for this flag that describes the behavior change?

And since this has a behavior change and a runtime feature flag, let's surface that to the current version history (https://github.com/envoyproxy/envoy/blob/main/docs/root/version_history/current.rst). As an example, you can find when the `envoy.reloadable_features.fix_added_trailers` was added here: https://github.com/envoyproxy/envoy/blob/1d1c63ceb19fd2420db13c7568a9302343e882de/docs/root/version_history/v1.20.0.rst#bug-fixes,